lib/modules/manager/pub/extract.ts

```diff
@@ -50,6 +59,7 @@ function extractFromSection(
       depType: sectionKey,
       currentValue,
       datasource: DartDatasource.id,
+      registryUrls,
```

Comment:
```suggestion
      ...(registryUrls && { registryUrls }),
```
only add if it has a value